model/src/main/kotlin/config/ScanStorageConfiguration.kt

```diff
@@ -115,3 +116,48 @@ data class PostgresStorageConfiguration(
      *       https://jdbc.postgresql.org/documentation/head/connect.html
      */
 ) : ScanStorageConfiguration()
+
+/**
+ * A class to hold the configuration for SW360.
+ */
+data class Sw360StorageConfiguration(
+    /**
+     * The REST API URL of SW360.
+     */
+    val restUrl: String,
+
+    /**
+     * The authentication URL of your SW360 instance.
+     */
+    val authUrl: String,
+
+    /**
+     * The username for the requests to SW360.
+     */
+    val username: String,
+
+    /**
+     * The password of the SW360 user.
+     */
+    val password: String,
+
+    /**
+     * The client ID of the SW360 instance for the two step authentication.
+     */
+    val clientId: String,
+
+    /**
+     * The password of the client ID.
+     */
+    val clientPassword: String,
+
+    /**
+     * The host of the proxy URL if one is in use.
+     */
+    val proxyHost: String? = null,
```

Comment:
What exactly is this used for? Should we use the proxy configuration known to ORT instead?

Reply:
To run this command behind a proxy, but you're right, it should take the ORT proxy configuration. Is there an easy way to obtain proxy configuration in the command?

Reply:
ORT configures a standard Java [ProxySelector](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/ProxySelector.html). Basically, this means there is no single correct proxy to use, as the proxy to use might depend on the URL you want to connect to.

@oheger-bosch what happens if we leave this at `null`? Would the sw360client (or the underlying HTTP client) automatically use the Java ProxySelector mechanism then?

Reply:
No, in this case a direct internet connection would be used.

Reply:
Checked again the code in the SW360 client library and the documentation of the underlying OkHttpClient. If no proxy settings are passed in the configuration of the SW360 Http client, then OkHttpClient will actually per default use the system proxy selector. So this is the desired behavior, correct?

So the SW360 client library is indeed missing a setting to enforce a direct Internet connection. You can either specify a concrete proxy host or fall back to the system proxy selector.

Reply:
> OkHttpClient will actually per default use the system proxy selector. So this is the desired behavior, correct?

Correct. BTW, here's the related ORT code that I wanted to point to: https://github.com/oss-review-toolkit/ort/blob/65082186f03861592d0955712d999b2af2a3763a/utils/src/main/kotlin/OkHttpClientHelper.kt#L67-L78

> So the SW360 client library is indeed missing a setting to enforce a direct Internet connection. You can either specify a concrete proxy host or fall back to the system proxy selector.

But that's fine, I believe. If you want a direct connection then you simply do not set a global proxy selector, or one that does not return proxy settings for the SW360 instance's URL.

Reply:
So bottom line, we should be able to remove `proxyHost` and `proxyPort` from here and hard-code `null` for these fields to make the sw360client library use ORT's proxy selector (which ORT sets globally), correct?

Reply:
Yes, exactly. The configuration options related to the proxy can be removed. When creating the configuration for the SW360 HttpClient, just do not set any proxy settings; then the default proxy selector is used.